modules/api/src/main/java/org/apache/ignite/schema/ColumnType.java

```diff
@@ -105,15 +105,55 @@ public static VarLenColumnType blobOf(int length) {
         return new VarLenColumnType(ColumnTypeSpec.BLOB, length);
     }
 
+    /**
+     * Number type factory method.
+     *
+     * @param precision Precision of value.
+     * @return Number type.
+     */
+    public static NumberColumnType numberOf(int precision) {
+        if (precision <= 0)
+            throw new IllegalArgumentException("Precision [" + precision + "] must be positive integer value.");
+
+        return new NumberColumnType(ColumnTypeSpec.NUMBER, precision);
+    }
+
+    /**
+     * Number type factory method.
+     *
+     * @return Number type.
+     */
+    public static NumberColumnType numberOf() {
+        return new NumberColumnType(ColumnTypeSpec.NUMBER, NumberColumnType.UNDEFINED);
+    }
+
     /**
      * Decimal type factory method.
      *
      * @param precision Precision.
      * @param scale Scale.
-     * @return Numeric type.
+     * @return Decimal type.
+     */
+    public static DecimalColumnType decimalOf(int precision, int scale) {
+        if (precision <= 0)
+            throw new IllegalArgumentException("Precision [" + precision + "] must be positive integer value.");
+        if (scale < 0)
+            throw new IllegalArgumentException("Scale [" + scale + "] must be non-negative integer value.");
```

Comment:
Just non-negative or strictly positive ?

Reply:
Precision should be Strictly positive value. It has no sense to store value with zero length. 
Scale should be non-negative. I think it is ok and means that we will store only Integer part of value.